### ISSUE TITLE
add worker panic when testing 500 jobs 50 worker per job in 10 nodes

### ISSUE THREAD

**Issue by hanfei1991** [hanfei1991/microcosm#267]
PANIC log
```
panic: runtime error: invalid memory address or nil pointer dereference[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x37b3170]goroutine 1080 [running]:github.com/hanfei1991/microcosm/lib.(*workerManagerImpl).addWorker.func1()        /root/microcosm/lib/worker_manager.go:454 +0x510github.com/pingcap/tiflow/pkg/workerpool.(*asyncWorker).run(0xc000f1f050)
        /root/go/pkg/mod/github.com/pingcap/tiflow@v0.0.0-20220308162507-e342c02e789c/pkg/workerpool/async_pool_impl.go:166 +0x23
github.com/pingcap/tiflow/pkg/workerpool.(*defaultAsyncPoolImpl).Run.func2()        /root/go/pkg/mod/github.com/pingcap/tiflow@v0.0.0-20220308162507-e342c02e789c/pkg/workerpool/async_pool_impl.go:114 +0x2a
golang.org/x/sync/errgroup.(*Group).Go.func1()        /root/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
        /root/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x8d
```

the culprit is worker d6ad8e53 
```
[2022/03/29 12:23:53.278 +08:00] [INFO] [master.go:613] ["Worker dispatched"] [master-id=c30402a0-1432-4f18-9106-477ba3b6363c] [response="error_code:OK worker_id:\"d6ad8e53-dba7-41b0-851a-3c7f75bec006\" "]

[2022/03/29 12:23:59.626 +08:00] [WARN] [worker_manager.go:450] ["worker has been removed"] [master-id=c30402a0-1432-4f18-9106-477ba3b6363c] [worker-id=d6ad8e53-dba7-41b0-851a-3c7f75bec006]
```

Probably the aysnc pool schedule is very slow when pressure is high

**Comment by liuzix** [comment]
This issue has been resolved in the latest master branch by #257

**Comment by amyangfei** [comment]
closed by #257
